FreeRTOS/FreeRTOS-Plus-TCP
# New helper function: FreeRTOS_get_tx_base

Description
-----------
In a [post](https://forums.freertos.org/t/stm32h743-freertos-tcp-issue-with-zero-copy/15390)  on the FreeRTOS forum, [@arijav](https://forums.freertos.org/u/arijav) asked for a new function `FreeRTOS_get_tx_base()`.

The existing `FreeRTOS_get_tx_head()` can be used to obtain a pointer to the head position in the TX buffer. It implements "zero copy": after filling the buffer, `FreeRTOS_send()` can be called with the buffer address `NULL`. The memcpy will be skipped.

The new `FreeRTOS_get_tx_base()` will return a pointer to the TX buffer itself.

Note that both functions are classified as "advanced". For Javier it was very helpful: he could increase the performance.

Test Steps
-----------
I tested both functions by calling them before and after the first call to `FreeRTOS_send()`.

I decided that both functions should create the TX buffer in case it has not been created yet.


By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.
